src/Slist.hs

```diff
@@ -490,6 +491,21 @@ cycle sl@(Slist _ Infinity) = sl
 cycle Slist{..}             = infiniteSlist $ L.cycle sList
 {-# INLINE cycle #-}
 
+{- | @O(1)@. A lazy slist equivalent of Haskell's '[from..to]' notation:
+creates an slist of sequentially ordered values starting at 'from' and ending at 'to'.
+
+>>> fromRange 0 5
+Slist {sList = [0,1,2,3,4,5], sSize = Size 6}
+>>> fromRange 5 0
+Slist {sList = [], sSize = Size 0}
+>>> fromRange 'a' 'd'
+Slist {sList = "abcd", sSize = Size 4}
```

Comment:
Nice examples!
Could you also please add an example of the behaviour with equal `from` and `to` :slightly_smiling_face: